test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go

```diff
@@ -666,7 +666,183 @@ var _ = f.CLICoreDescribe("[Tests:E2E][Feature:Plugin-Sync-TMC-lifecycle]", func
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(f.CheckAllPluginsExists(installedPluginsListK8s, pluginsInfoForCRsApplied)).Should(BeTrue(), " plugins being installed and plugins info for which CRs applied should be same")
+		})
+
+		var pluginsToGenerateMockResponseTMC, installedPluginsListTMC []*f.PluginInfo
+		var contextNameTMC string
+		var ok bool
+
+		// Test case: e. TMC: mock tmc endpoint with plugins info, start the mock server
+		It("mock tmc endpoint with expected plugins response and restart REST API mock server", func() {
+			// get plugins from a group
+			pluginsToGenerateMockResponseTMC, ok = pluginGroupToPluginListMap[usePluginsFromTmcPluginGroup]
+			Expect(ok).To(BeTrue(), pluginGroupShouldExists)
+			Expect(len(pluginsToGenerateMockResponseTMC) > numberOfPluginsToInstall).To(BeTrue(), testRepoDoesNotHaveEnoughPlugins)
+			// mock tmc endpoint with only specific number of plugins info
+			pluginsToGenerateMockResponseTMC = pluginsToGenerateMockResponseTMC[:numberOfPluginsToInstall]
+			mockReqResMapping, err := f.ConvertPluginsInfoToTMCEndpointMockResponse(pluginsToGenerateMockResponseTMC[:numberOfPluginsToInstall])
+			Expect(err).To(BeNil(), noErrorForMockResponsePreparation)
+			err = f.WriteToFileInJSONFormat(mockReqResMapping, tmcPluginsMockFilePath)
+			Expect(err).To(BeNil(), noErrorForMockResponseFileUpdate)
+
+			// start http mock server
+			err = f.StartMockServer(tf, tmcConfigFolderPath, f.HttpMockServerName)
+			Expect(err).To(BeNil(), mockServerShouldStartWithoutError)
+			var mockResPluginsInfo f.TMCPluginsInfo
+			// check the tmc mocked endpoint is working as expected
+			err = f.GetHTTPCall(f.TMCPluginsMockServerEndpoint, &mockResPluginsInfo)
+			Expect(err).To(BeNil(), "there should not be any error for GET http call on mockapi endpoint:"+f.TMCPluginsMockServerEndpoint)
+			Expect(len(mockResPluginsInfo.Plugins)).Should(Equal(len(pluginsToGenerateMockResponseTMC)), "the number of plugins in endpoint response and initially mocked should be same")
+			totalInstalledPlugins += numberOfPluginsToInstall
+		})
+		// Test case: f. TMC: create context and make sure context has created
+		It("create context for TMC target with http mock server URL as endpoint", func() {
+			// Clean K8s context specific plugins
+			err = tf.PluginCmd.CleanPlugins()
+			Expect(err).To(BeNil(), "plugin clean should not return any error")
+
+			contextNameTMC = f.ContextPrefixTMC + f.RandomString(4)
+			_, _, err = tf.ContextCmd.CreateContextWithEndPointStaging(contextNameTMC, f.TMCMockServerEndpoint, f.AddAdditionalFlagAndValue(forceCSPFlag))
+			Expect(err).To(BeNil(), noErrorWhileCreatingContext)
+			active, err := tf.ContextCmd.GetActiveContext(string(types.TargetTMC))
+			Expect(err).To(BeNil(), activeContextShouldExists)
+			Expect(active).To(Equal(contextNameTMC), activeContextShouldBeRecentlyAddedOne)
+			contexts = append(contexts, contextNameTMC)
```

Comment:
Would it be possible to also test `tanzu context use` for the new single-context sync?

Reply:
The new test for `tanzu context use` that was added in a previous commit is gone

Reply:
i have added tests for `tanzu context use` for both tmc/k8s